roadmap/implementers-guide/src/node/disputes/participation.md

```diff
@@ -0,0 +1,94 @@
+# Dispute Participation
+
+Tracks all open disputes in the active leaves set.
+
+## Remarks
+
+`UnsignedTransaction`s are OK to be used, since the inner
+element `CommittedCanddidateReceipt` is verifiable.
+
+## ToDo
+
+1. Specify what is being gossiped, what messages and what they must contain.
+
+
+## IO
+
+Inputs:1. Receive `DisputeParticipationMessage::Resolution`
+1. Delete all relevant data associated with this dispute
+
+
+* `DisputeParticipationMessage::Detection`
+* `DisputeParticipationMessage::Resolution`
+
+Outputs:
+
+* `AvailabilityRecoveryMessage::RecoverAvailableData`
+* `CandidateValidationMessage::ValidateFromChainState`
+* `...::Blacklist`
+
+## Messages
+
+```rust
+enum DisputeParticipationMessage {
+    Detection {
+        candidate: CandidateHash,
+        votes: HashMap<ValidatorId, Vote>,
+    },
+
+    Resolution{
+        resolution: Resolution,
+        votes: HashMap<ValidatorId, Vote>,
+    },
+}
+```
+
+## Helper Structs
+
+```rust
+/// Resolution of a vote:
+enum Resolution {
+    /// The block is valid
+    Valid,
+    /// The block is valid
+    Invalid,
+}
+```
+
+## Session Change
+
+> TODO
+
+## Storage
+
+Nothing to store, everything lives in `VotesDB`.
+
+## Sequence
+
+### Resolution
+
+1. Receive `DisputeParticipationMessage::Detection`
+1. Request `AvailableData` via `RecoverAvailableData` to obtain the `PoV`.
+1. Call `ValidateFromChainState` to validate the `PoV` with the `CandidateDescriptor`.
+    1. Cast vote according to the `ValidationResult`.
+    1. Store our vote to the `VotesDB` via `VotesDBMessage::StoreVote`
+    1. Gossip our vote on the network.
+
+### Resolution
+
+1. Receive `DisputeParticipationMessage::Resolution`
+1. Craft an unsigned transaction with all votes cast
+  1. includes `CandidateReceipts`
+  1. includes `Vote`s
+1. Delete all relevant data associated with this dispute
+
+### Blacklisting bad block and invalid decendants
+
+In case a block was disputed successfully, and is now deemed invalid.
+
+1. Query `ChainApiMessage::Descendants` for all descendants of the disputed `Hash`.
+1. Blacklist all descendants of the disputed block `ChainApiMessage::Blacklist`.
+1. Check if the dispute block was finalized
+1. iff:
+    1. put the chain into governance mode
```

Comment:
I don't think "governance mode" has been formally defined anywhere. Pretty sure it just means "halt the chain until restarted by some entity with superuser permissions", but searching the current impl of the guide doesn't give me that definition.

Reply:
I think I left a TODO for exactly that at the very end :)

Reply:
It likely means pausing all parachains and maybe halts transactions on the relay chain too.  In the short term, we mostly just want to minimize our own headache whenever we cause this accidentally like by some mistake in a wasm upgrade or whatever.  ;)

Reply:
For me this is more about the mechanics, of how to halt the chain while technically staying operational such that sudo/governance decisions can still "resolve it", which implies halting the relay chain does not work, but halting all para chains should be fine™.

Given that this is very fuzzy, I'd be in favour of extracting that into a separate PR.

Reply:
It's definitely a runtime concern. The way that we would implement this in practice is by disabling all parachain import from this point onwards by setting a flag in the runtime. i.e. it would happen automatically from the perspective of the node-side code.